flex_container_orchestrator/domain/aggregator_flexpart.py

```diff
@@ -0,0 +1,340 @@
+#!/usr/bin/env python3
+
+import argparse
+import datetime
+import json
+import logging
+import sqlite3
+import sys
+from typing import List, Optional, Set, Tuple
+
+from flex_container_orchestrator import CONFIG
+
+
+def parse_arguments() -> argparse.Namespace:
+    """
+    Parse command-line arguments.
+
+    Returns:
+        argparse.Namespace: Parsed arguments.
+    """
+    parser = argparse.ArgumentParser(
+        description="Launch Flexpart after processing checks."
+    )
+    parser.add_argument("--date", required=True, help="Date in YYYYMMDD format")
+    parser.add_argument("--time", required=True, help="Time in HH format")
+    parser.add_argument(
+        "--step", required=True, help="Step identifier (lead time in hours)"
+    )
+    parser.add_argument("--db_path", required=True, help="Path to the SQLite database")
+    return parser.parse_args()
+
+
+def connect_db(db_path: str) -> sqlite3.Connection:
+    """
+    Establish a connection to the SQLite database.
+
+    Args:
+        db_path (str): Path to the SQLite database.
+
+    Returns:
+        sqlite3.Connection: SQLite connection object.
+    """
+    try:
+        conn = sqlite3.connect(db_path)
+        logging.info("Connected to SQLite database at %s.", db_path)
+        return conn
+    except sqlite3.Error as e:
+        logging.error("SQLite connection error: %s", e)
+        sys.exit(1)
+
+
+def is_row_processed(
+    conn: sqlite3.Connection, forecast_ref_time: datetime.datetime, step: str
+) -> bool:
+    """
+    Check if a specific row in the database has been processed.
+
+    Args:
+        conn (sqlite3.Connection): SQLite connection object.
+        forecast_ref_time (datetime): Forecast reference time
+        step (str): Step identifier.
+
+    Returns:
+        bool: True if processed, False otherwise.
+    """
+    try:
+        cursor = conn.cursor()
+        cursor.execute(
+            """
+            SELECT processed FROM uploaded
+            WHERE forecast_ref_time = ? AND step = ?
+        """,
+            (forecast_ref_time, step),
+        )
+        result = cursor.fetchone()
+        if result:
+            return result[0] == 1
+        logging.info(
+            "No row found for forecast_ref_time=%s and step=%s.",
+            forecast_ref_time,
+            step,
+        )
+        return False
+    except sqlite3.Error as e:
+        logging.error("SQLite query error: %s", e)
+        sys.exit(1)
+
+
+def generate_flexpart_start_times(
+    frt_dt: datetime.datetime, lead_time: int, tdelta: int, tfreq_f: int
+) -> List[datetime.datetime]:
+    """
+    Generate a list of Flexpart run start times.
```

Comment:
I think this docstring could explain better because I still need to read the whole function to understand what it is doing.